### PR TITLE
[libc][math][c23] Add entrypoints and tests for totalorder{,f,f128}

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by overmighty** [comment]
I'm not sure if you've used https://github.com/overmighty/llvm-dev-utils/blob/master/llvm_libc_add_math_function.py or not since the libc/src/math/totalorderf.h header has an extra character.

Here's how you could have used it:

```sh
for type in '' f l f128; do
  ./replace/with/path/to/llvm_libc_add_math_function.py "${type}" 'TYPE' 'totalorder' 'const TYPE *x, const TYPE *y' 'TotalOrderTest' 'LIST_TOTALORDER_TESTS'
done
```

For example:

```sh
for type in '' f l f128; do
  ~/projects/llvm-dev-utils/llvm_libc_add_math_function.py "${type}" 'TYPE' 'totalorder' 'const TYPE *x, const TYPE *y' 'TotalOrderTest' 'LIST_TOTALORDER_TESTS'
done
```

**Comment by overmighty** [comment]
I'm guessing x86 `long double` needs special handling in `fputil::totalorder`, since only the `totalorderl` tests are failing.

**Comment by Jobhdez** [comment]
> I'm guessing x86 `long double` needs special handling in `fputil::totalorder`, since only the `totalorderl` tests are failing.

what do you mean "special handling"?

**Comment by overmighty** [comment]
The x86 80-bit extended precision format is a bit more complicated than the IEEE binary formats. Maybe the current logic in `fputil::totalorder` doesn't work for the x86 80-bit format, I haven't looked into why exactly the `totalorderl` tests are failing.

File: libc/docs/math/index.rst [hunk 1]
```diff
@@ -228,7 +228,7 @@ Basic Operations
 +------------------+------------------+-----------------+------------------------+----------------------+------------------------+------------------------+----------------------------+
 | setpayloadsig    |                  |                 |                        | |check|              |                        | F.10.13.3              | N/A                        |
 +------------------+------------------+-----------------+------------------------+----------------------+------------------------+------------------------+----------------------------+
-| totalorder       |                  |                 |                        | |check|              |                        | F.10.12.1              | N/A                        |
+| totalorder       | |check|          | |check|         | |check|                | |check|              | |check|                | F.10.12.1              | N/A                        |
```
Comment (overmighty) [inline libc/docs/math/index.rst:231]: The check mark in the `long double` column should be removed too.

File: libc/src/math/totalorderf.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+//===-- Implementation header for totalorderf --------------------*- C++
+//-*-===//
```
Comment (overmighty) [inline libc/src/math/totalorderf.h:2]: Nit:

```suggestion
//===-- Implementation header for totalorderf -------------------*- C++ -*-===//
```

**Review by overmighty — APPROVED** [review]

**Comment by mikhailramalho** [comment]
This PR is causing a build failure on the rv32 buildbot: https://lab.llvm.org/staging/#/builders/132

**Comment by Jobhdez** [comment]
> I'll fix it.

Thanks

**Pull request merged**